hugr-py/src/hugr/std/collections/array.py

```diff
@@ -60,7 +60,7 @@ class ArrayVal(val.ExtensionValue):
     """Constant value for a statically sized array of elements."""
 
     v: list[val.Value]
-    ty: tys.Type
+    ty: Array
```

Comment:
this is non-breaking because `Array` satisfies the Type protocol, correct?

Reply:
It's fine for reading since it's a type refinement.

For _writing_ into the variable it is actually a breaking change. I assumed the dataclass was frozen, but it seems not.

Anyways I'd say mutating the attribute directly is not common (since the type is normally constructed using its custom `__init__`), and putting in a non-array type should be a bug anyways.